x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/shared/map_processor_type_to_form.tsx

```diff
@@ -246,210 +371,373 @@ export const mapProcessorTypeToDescriptor: MapProcessorTypeToDescriptor = {
+        values: {
+          field,
+        },
+      }),
   },
   pipeline: {
     FieldsComponent: Pipeline,
     docLinkPath: '/pipeline-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.pipeline', {
       defaultMessage: 'Pipeline',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.pipeline', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.pipeline', {
       defaultMessage: 'Runs another ingest node pipeline.',
     }),
+    getDefaultDescription: ({ name }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.pipeline', {
+        defaultMessage: 'Executes pipeline "{name}"',
+        values: {
+          name,
+        },
+      }),
   },
   remove: {
     FieldsComponent: Remove,
     docLinkPath: '/remove-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.remove', {
       defaultMessage: 'Remove',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.remove', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.remove', {
       defaultMessage: 'Removes one or more fields.',
     }),
+    getDefaultDescription: ({ field }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.remove', {
+        defaultMessage: 'Removes field(s) "{field}"',
+        values: {
+          field: Array.isArray(field as unknown)
+            ? ((field as unknown) as string[]).join(',')
+            : field,
+        },
+      }),
   },
   rename: {
     FieldsComponent: Rename,
     docLinkPath: '/rename-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.rename', {
       defaultMessage: 'Rename',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.rename', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.rename', {
       defaultMessage: 'Renames an existing field.',
     }),
+    getDefaultDescription: ({ field, target_field: targetField }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.rename', {
+        defaultMessage: 'Renames "{field}" to "{target_field}"',
+        values: {
+          field,
+          target_field: targetField,
+        },
+      }),
   },
   script: {
     FieldsComponent: Script,
     docLinkPath: '/script-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.script', {
       defaultMessage: 'Script',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.script', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.script', {
       defaultMessage: 'Runs a script on incoming documents.',
     }),
+    // TODO: Decide whether requires user to provide a meaningful description
+    getDefaultDescription: () => undefined,
```

Comment:
Missing a description here. I'd suggest using the `defaultMessage` from line 574.